app/content/react/guides/query-parameters.md

```diff
@@ -0,0 +1,124 @@
+## Query Parameters
+
+You can use QueryParameters to filter your layers & widgets at the backend side, using the concept of *parametrized queries* over your sources. You can check our related [deck.gl documentation]({{< ref "/deck-gl/reference#queryparameters-depends-on-provider-optional" >}} "About Us") for different providers.
+In this guide we're going to create a simple map that shows the earthquakes in Spain and allows to filter it by their magnitude.
+
+#### Creating a map using QueryParameters
+
+First you'll need to create a source that supports query parameters. In this case we are getting all the earthquakes whoses magnitude are between `min: 4` and `max: 5`.
+
+```ts
+import { MAP_TYPES } from '@deck.gl/carto';
+
+const sql = 'select * from `carto-demo-data.demo_tables.spain_earthquakes` where magnitude between @min and @max';
+
+const earthquakesSource = {
+  id: 'earthquakes',
+  data: sql,
+  type: MAP_TYPES.QUERY,
+  queryParameters: {
+    min: 4,
+    max: 5
+  }
+}
+```
+
+Once you have your source ready to be used, you can create your layer using `useCartoLayerProps` hook.  
+
+```ts
+import { CartoLayer } from '@deck.gl/carto';
+import { useCartoLayerProps } from '@carto/react-api';
+
+const layerProps = useCartoLayerProps({ source: earthquakesSource });
+const earthquakesLayer = new CartoLayer({
+  ...layerProps,
+  id: 'earthquakesLayer',
+  pointRadiusUnits: 'pixels',
+  lineWidthUnits: 'pixels',
+  getLineColor: [0, 0, 0],
+  getPointRadius: 3,
+});
+```
+
+Now your layer is ready and you can use it in your map, so we're going to create a `<FormulaWidget />` which displays the number of earthquakes in your viewport.
+
+#### Using queryParameters with widgets
+All our widgets are ready to use queryParameters. You just need to pass the `datasource` properly configured (that basically means declaring its `type` as MAP_TYPES.QUERY and adding the corresponding `queryParameters` configuration). After that, you can use any widget as usual, eg:
```

Comment:
This sentence `All our widgets are ready to use queryParameters` sounds confusing to me. Probably I would say
All our widgets are ready to use **datasources** with queryParameters

Reply:
Much better